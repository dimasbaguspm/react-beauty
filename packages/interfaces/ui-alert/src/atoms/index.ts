export * from './alert-container';
export * from './alert-description';
export * from './alert-icon';
export * from './alert-title';
export * from './alert-dismiss-button';
