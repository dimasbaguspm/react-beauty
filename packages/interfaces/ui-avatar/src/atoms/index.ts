export * from './avatar-container';
export * from './avatar-initial';
export * from './avatar-picture';
