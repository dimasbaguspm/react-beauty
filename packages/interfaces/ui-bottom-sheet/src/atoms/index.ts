export * from './bottom-sheet-body';
export * from './bottom-sheet-container';
export * from './bottom-sheet-footer';
export * from './bottom-sheet-header';
export * from './bottom-sheet-title';
export * from './bottom-sheet-close-button';
export * from './style';
export * from './use-bottom-sheet';
