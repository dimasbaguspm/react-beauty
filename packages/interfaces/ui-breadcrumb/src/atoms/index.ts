export * from './breadcrumb-container';
export * from './breadcrumb-list';
export * from './breadcrumb-item';
export * from './breadcrumb-link';
export * from './breadcrumb-divider';
