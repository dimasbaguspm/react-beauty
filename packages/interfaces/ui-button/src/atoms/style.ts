import { styled } from "@linaria/react";

const fill = `
  background-color: var(--components-button-colors-fill-default-bg);
  color: var(--components-button-colors-fill-default-color);

  &:hover {
    background-color: color-mix(in srgb, var(--colors-main-picollo), var(--colors-main-bulma) 10%);
  }

  &:focus-visible {
    box-shadow: 0 0 0 4px var(--components-button-colors-fill-focus-bs);
  }
 `;

const outline = `
  outline: solid;
  outline-color: var(--components-button-colors-outline-default-outline);
  outline-width: 1px;
  color: var(--components-button-colors-outline-default-color);
  background-color: transparent;

  &:hover {
    background-color: var(--components-button-colors-outline-hover-bg);
  }

  &:focus-visible {
    box-shadow: 0 0 0 4px var(--components-button-colors-outline-focus-bs);
  }
`;

const ghost = `
  background-color: transparent;
  color: var(--components-button-colors-ghost-default-color);

  &:hover {
    color: var(--components-button-colors-ghost-hover-color);
    background-color: var(--components-button-colors-ghost-hover-bg);
  }

  &:focus-visible {
    box-shadow: 0 0 0 4px color-mix(in srgb, var(--colors-main-picollo) 20%, transparent);
  }
`;

const sizeXs = `
  border-radius: var(--components-button-border-radius-xs);
  padding: var(--components-button-padding-xs-y) var(--components-button-padding-xs-x);
  font-size: var(--components-button-font-size-xs);
  font-weight: var(--components-button-font-weight);
  line-height: var(--components-button-line-height-xs);

  &[data-is-only-icon="true"] {
    padding: var(--components-button-padding-xs-icon);
  }
`;

const sizeSm = `
  border-radius: var(--components-button-border-radius-sm);
  padding: var(--components-button-padding-sm-y) var(--components-button-padding-sm-x);
  font-size: var(--components-button-font-size-sm);
  font-weight: var(--components-button-font-weight);
  line-height: var(--components-button-line-height-sm);

  &[data-is-only-icon="true"] {
    padding: var(--components-button-padding-sm-icon);
  }
`;

const sizeMd = `
  border-radius: var(--components-button-border-radius-md);
  padding: var(--components-button-padding-md-y) var(--components-button-padding-md-x);
  font-size: var(--components-button-font-size-md);
  font-weight: var(--components-button-font-weight);
  line-height: var(--components-button-line-height-md);

  &[data-is-only-icon="true"] {
    padding: var(--components-button-padding-md-icon);
  }
`;

const sizeLg = `
  border-radius: var(--components-button-border-radius-lg);
  padding: var(--components-button-padding-lg-y) var(--components-button-padding-lg-x);
  font-size: var(--components-button-font-size-lg);
  font-weight: var(--components-button-font-weight);
  line-height: var(--components-button-line-height-lg);

  &[data-is-only-icon="true"] {
    padding: var(--components-button-padding-lg-icon);
  }
`;

const sizeXl = `
  border-radius: var(--components-button-border-radius-xl);
  padding: var(--components-button-padding-xl-y) var(--components-button-padding-xl-x);
  font-size: var(--components-button-font-size-xl);
  font-weight: var(--components-button-font-weight);
  line-height: var(--components-button-line-height-xl);

  &[data-is-only-icon="true"] {
    padding: var(--components-button-padding-xl-icon);
  }
`;

export const ElButton = styled.button`
  outline: none;
  border: none;
  background-color: unset;

  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: var(--space-1);

  transition: transform 0.2s ease-in-out;

  &:hover {
    cursor: pointer;
  }

  &:active {
    transform: scale(0.9);
  }

  &[disabled] {
    cursor: not-allowed;
    opacity: 32%;
  }

  &[data-is-full-width] {
    width: 100%;
  }

  &[data-size="xs"] {
    ${sizeXs}
  }

  &[data-size="sm"] {
    ${sizeSm}
  }

  &[data-size="md"] {
    ${sizeMd}
  }

  &[data-size="lg"] {
    ${sizeLg}
  }

  &[data-size="xl"] {
    ${sizeXl}
  }

  &[data-variant="fill"] {
    ${fill}
  }

  &[data-variant="outline"] {
    ${outline}
  }

  &[data-variant="ghost"] {
    ${ghost}
  }
`;
