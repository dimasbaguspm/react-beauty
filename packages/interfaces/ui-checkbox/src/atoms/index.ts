export * from './checkbox-container';
export * from './checkbox-input';
export * from './checkbox-input-label';
export * from './checkbox-label';
export * from './checkbox-group';
export * from './checkbox-group-helper-text';
export * from './checkbox-input-field';
export * from './use-checkbox';
export * from './use-checkbox-group';
