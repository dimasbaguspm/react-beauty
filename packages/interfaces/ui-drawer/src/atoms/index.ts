export * from './drawer-body';
export * from './drawer-container';
export * from './drawer-dismiss-button';
export * from './drawer-footer';
export * from './drawer-header';
export * from './drawer-title';
