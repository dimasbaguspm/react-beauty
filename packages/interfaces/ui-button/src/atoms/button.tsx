import { ButtonHTMLAttributes, forwardRef, ReactNode } from 'react';

import { ButtonSize, ButtonVariant } from '../type';

import { ElButton } from './style';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  onClick: VoidFunction;
  variant?: ButtonVariant;
  size?: ButtonSize;
  isFullWidth?: boolean;
  prefixIcon?: ReactNode;
  suffixIcon?: ReactNode;
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      variant = 'fill',
      size = 'md',
      isFullWidth,
      children,
      prefixIcon,
      suffixIcon,
      ...props
    },
    ref,
  ) => {
    return (
      <ElButton
        {...props}
        data-variant={variant}
        data-size={size}
        data-is-full-width={isFullWidth}
        data-is-only-icon={typeof children !== 'string'}
        ref={ref}
      >
        {Boolean(prefixIcon) && prefixIcon}
        {children}
        {Boolean(suffixIcon) && suffixIcon}
      </ElButton>
    );
  },
);
