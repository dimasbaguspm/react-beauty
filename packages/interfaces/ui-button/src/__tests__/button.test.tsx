import { act, fireEvent, render, screen } from '@react-beauty/vitest/selector';

import { Button } from '../atoms';

describe('Button', () => {
  it('should render correctly', () => {
    expect(
      render(<Button onClick={vi.fn()}>Button</Button>).asFragment(),
    ).toMatchSnapshot();
  });

  it('should follow the standard aria button', () => {
    const mockClickFn = vi.fn();

    render(<Button onClick={mockClickFn}>Button</Button>);

    act(() => {
      fireEvent.click(screen.getByRole('button'));
    });

    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(mockClickFn).toHaveBeenCalledTimes(1);
  });
});
