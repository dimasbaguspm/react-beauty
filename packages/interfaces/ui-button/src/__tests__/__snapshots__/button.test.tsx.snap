// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render correctly 1`] = `
<DocumentFragment>
  <button
    class="mocked-styled-0"
    data-is-only-icon="false"
    data-size="sm"
    data-variant="fill"
  >
    Button
  </button>
</DocumentFragment>
`;
